EpiAware/src/epimodel.jl

```diff
@@ -61,7 +61,7 @@ struct DirectInfections <: AbstractEpiModel
 end
 
 function (epimodel::DirectInfections)(_It, latent_process_aux)
-    epimodel.data.transformation.(_It)
+    epimodel.data.transformation.(latent_process_aux.init .+ _It)
```

Comment:
Noting that passing in a list  feels like a bit of a bad pattern here. I would usually have thought specifiying the argumments explicitly and then calling the list of args on the function would be a better approach?

Reply:
The reason for using this pattern is that its flexible enough to be represented as:

https://github.com/CDCgov/Rt-without-renewal/blob/2e18d00529a716226088a2df1d1016059b884385/EpiAware/src/models.jl#L13-L16

for any combination of latent process, and method for going from latent process -> actual infections.

At the moment, we are assuming that the generative distribution for initialisation "lives" inside the latent process as a random value `init` among whatever auxillary values are generated by the latent process. I could see an argument for this being moved elsewhere.

Passing a `NamedTuple` is pretty common, I don't think its a bad pattern. For example, in [LogDensityProblems.jl](https://github.com/tpapp/LogDensityProblems.jl) the standard pattern is to define a data struct contain the fixed data of an inference problem with a callable function taking a `NamedTuple` arg and outputing the lpd.

Reply:
So we could instead use `...args` here? The advantage of that is then all the input requirements of a specific latent process would be clearly documented within the function specification. 

If you test and document the `NamedTuple` input requirements then this is less of an issue but I've seen it go wrong a fair few times (and people just forget what inputs functions are using etc).

Reply:
To be clear I am not opposed to this but just stating some potential issues. I think if we move to a struct for submodules this becomes less of an issue as the object that contains the function will be unique vs being a generic shared prior structure.

Reply:
> To be clear I am not opposed to this but just stating some potential issues. I think if we move to a struct for submodules this becomes less of an issue as the object that contains the function will be unique vs being a generic shared prior structure.

I think this is the way to go ala #45 .

Reply:
> So we could instead use `...args` here? The advantage of that is then all the input requirements of a specific latent process would be clearly documented within the function specification.
> 
> If you test and document the `NamedTuple` input requirements then this is less of an issue but I've seen it go wrong a fair few times (and people just forget what inputs functions are using etc).

I think the callable function from `EpiModel` subtype to actual infections only needs 
1. A latent process input
2. An `init` value (which will virtually always be unconstrained/transformed time zero incidence).

Reply:
I'm going to alter the pattern to reflect that.

Reply:
I've just pushed a minor change where the `init` value is:
- Directly outputted by the latent process
- used as a direct arg in `epimodel`

rather than in both cases being "hidden" as the optional field of a `NamedTuple` object (with the potential gotchas you mention).

Doesn't change the reasons for #45 though.